library/std/src/sys/unix/ext/fs.rs

```diff
@@ -345,6 +345,33 @@ pub trait OpenOptionsExt {
     /// ```
     #[stable(feature = "open_options_ext", since = "1.10.0")]
     fn custom_flags(&mut self, flags: i32) -> &mut Self;
+
+    /// Get the flags of this OpenOptions as libc::c_int.
+    ///
+    /// This method allows the reuse of the OpenOptions as flags argument for `libc::open()`.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # #![feature(rustc_private)]
+    /// extern crate libc;
+    /// use std::ffi::CString;
+    /// use std::fs::OpenOptions;
+    /// use std::os::unix::fs::OpenOptionsExt;
+    ///
+    /// # fn main() {
+    /// let mut options = OpenOptions::new();
+    /// options.write(true).read(true);
+    /// if cfg!(unix) {
+    ///     options.custom_flags(libc::O_NOFOLLOW);
+    /// }
+    /// let file_name = CString::new("foo.txt").unwrap();
+    /// let file = unsafe{ libc::open(file_name.as_c_str().as_ptr(), options.as_flags().unwrap()) };
+    ///
+    /// # }
+    /// ```
+    #[stable(feature = "open_options_ext_as_flags", since = "1.47.0")]
+    fn as_flags(&self) -> io::Result<libc::c_int>;
```

Comment:
Is there any reason `libc::c_int` is used instead of `std::os::raw::c_int`?

Reply:
Just because both `get_access_mode` and `get_creation_mode` use `libc::c_int`. Shall I change the return type to `std::os::raw::c_int` instead?

Reply:
No, I didn't saw the rest.